dask_jobqueue/core.py

```diff
@@ -607,6 +625,46 @@ def _new_worker_name(self, worker_number):
             cluster_name=self._name, worker_number=worker_number
         )
 
+    def _get_worker_security(self, security):
+        """Dump temporary parts of the security object into a shared_directory"""
+        if security is not None:
+            copied_on_write = False
+            worker_security_dict = security.get_tls_config_for_role("worker")
+            for key, value in worker_security_dict.items():
+                # dump worker in-memory keys for use in job_script
+                if value is not None and "\n" in value:
+                    if not copied_on_write:
+                        security = copy.copy(security)
+                        copied_on_write = True
+                    f = tempfile.NamedTemporaryFile(
+                        mode="wt",
+                        prefix=".dask-jobqueue.worker." + key + ".",
+                        dir=(
+                            os.path.expanduser(
+                                os.path.expandvars(self.shared_directory)
+                            )
+                            if self.shared_directory is not None
+                            else None
```

Comment:
What is the behavior if dir is None here?

Reply:
Actually I will rather assert that this is not None by using a default. Then this is more clear.